entrypoint/versionbump/componentsbump_test.go

```diff
@@ -0,0 +1,177 @@
+			assert.NodeAssertReadyStatus(),
+			nil)
+	})
+
+	It("Validate Pod", func() {
+		testcase.TestPodStatus(
+			assert.PodAssertRestart(),
+			assert.PodAssertReady(),
+			assert.PodAssertStatus())
+	})
+
+	var runc = fmt.Sprintf("(find /var/lib/rancher/%s/data/ -type f -name runc -exec {} --version \\;)", cfg.Product)
+
+	// test decription and cmds generated based on product rke2
+	description := "Verifies bump versions for several components on Rke2:\n1-canal\n2-flannel\n" +
+		"3-calico\n4-ingressController\n5-coredns\n6-metricsServer\n7-etcd\n8-containerd\n9-runc"
+	cmd := flannelRke2 + calico + ingressController + corednsRke2 + metricsServer + etcdRke2 + containerd + runc
+
+	// test decription and cmds updated based on product k3s
+	if cfg.Product == "k3s" {
+		description = "Verifies bump versions for several components on k3s:\n1-flannel\n2-coredns\n3-metricsServer\n" +
+			"4-etcd\n5-cni plugins\n6-traefik\n7-local path storage\n8-containerd\n9-Klipper\n10-runc"
+		cmd = flannelK3s + coreDnsk3s + metricsServer + etcdK3s + cniPlugins + traefik + localPath + containerd + klipperLB + runc
+	}
+
+	It(description, func() {
+		Template(TestTemplate{
+			TestCombination: &RunCmd{
+				Run: []TestMap{
+					{
+						Cmd:                  cmd,
+						ExpectedValue:        TestMapTemplate.ExpectedValue,
+						ExpectedValueUpgrade: TestMapTemplate.ExpectedValueUpgrade,
+					},
+				},
+			},
+			InstallMode: customflag.ServiceFlag.InstallMode.String(),
+			Description: customflag.ServiceFlag.TestConfig.Description,
+			DebugMode:   customflag.ServiceFlag.TestConfig.DebugMode,
+		})
+	})
+
+	It("Verifies dns access", func() {
+		testcase.TestDnsAccess(true, false)
+	})
+
+	It("Verifies Ingress", func() {
+		testcase.TestIngress(true, true)
+	})
+
+	// we start tests here with cni: multus+canal
+	if cfg.Product == "rke2" {
+		It("Updating config yaml cni to cilium and get version bumps for it and cni-plugins", func() {
+			Template(TestTemplate{
+				TestCombination: &RunCmd{
+					Run: []TestMap{
+						{
+							Cmd:           "kubectl get node -o yaml : | grep multus-cni -A1",
+							ExpectedValue: "hardened-multus ",
+						},
+						{
+							Cmd: "sudo sed -i '/cni:/d' /etc/rancher/rke2/config.yaml && " +
+								" sudo sed -i '/multus/d' /etc/rancher/rke2/config.yaml && sudo sed -i '/canal/d' /etc/rancher/rke2/config.yaml " +
+								" && echo -e \"cni: cilium\" | sudo tee -a /etc/rancher/rke2/config.yaml " +
+								" && cat /etc/rancher/rke2/config.yaml",
+							ExpectedValue: "cilium",
+						},
+						{
+							Cmd:           "sudo sed -i '/^$/d' /etc/rancher/rke2/config.yaml && cat /etc/rancher/rke2/config.yaml && sleep 20",
+							ExpectedValue: "cilium",
+						},
+						{
+							Cmd:           "sudo systemctl restart rke2-server && if [ $? -eq 0 ]; then echo \"ok\"; fi",
+							ExpectedValue: "ok",
+						},
+						{
+							Cmd:           "sleep 10 && if [ $? -eq 0 ]; then echo \"ok\"; fi",
+							ExpectedValue: "ok",
```

Comment:
Could we add sleep with the restart cmd instead of having it as a separate cmd?

Reply:
this one i would like to keep separate just to make more clear, like if we have any error from restart we will know thats from there , rather than go to logs to find out if the "  nok " answer is from restart or from some other network/connection issue that might happen when trying to sleep 

but its just to separate concerns nothing else .